CMakeLists.txt

```diff
@@ -154,40 +154,58 @@ if(ENABLE_COVERAGE)
 endif(ENABLE_COVERAGE)
 
 #
-# Configure which bridge device to use with SC
+# Network related options
 #
 
-if(NOT DEFINED BRIDGE_DEVICE)
-    set(BRIDGE_DEVICE lxcbr0)
-endif()
-add_definitions(-DBRIDGE_DEVICE="${BRIDGE_DEVICE}")
-message("Network bridge defined as ${BRIDGE_DEVICE}")
+if (ENABLE_NETWORKGATEWAY)
 
-if(NOT DEFINED BRIDGE_DEVICE)
-    set(BRIDGE_DEVICE lxcbr0)
-endif()
-add_definitions(-DBRIDGE_DEVICE="${BRIDGE_DEVICE}")
-message("Network bridge defined as ${BRIDGE_DEVICE}")
+    macro(set_default VAR VALUE)
+        if(NOT DEFINED ${VAR})
+            set(${VAR} ${VALUE})
+        endif()
+    endmacro()
 
-#
-# If SC shouldn't create bridge, then we substitute different parts
-# of the setup script at compile time.
-#
+    set_default(BRIDGE_DEVICE lxcbr0)
+    add_definitions(-DBRIDGE_DEVICE="${BRIDGE_DEVICE}")
+    message(STATUS "Network bridge defined as ${BRIDGE_DEVICE}")
 
-if(CREATE_BRIDGE)
-    message("SoftwareContainer will create ${BRIDGE_DEVICE} if not present at runtime")
-    file(READ scripts/setup_create_nobridge.sh.in NOBRIDGE)
-else()
-    message("SoftwareContainer will exit if ${BRIDGE_DEVICE} is not present at runtime")
-    file(READ scripts/setup_exit_nobridge.sh.in NOBRIDGE)
-endif()
+    #
+    # Should SoftwareContainer create a bridge if not present?
```

Comment:
Nitpicking: this is almost phrased as a TODO, it could perhaps be more clear that we know what we want to check and how to behave.